runtime/interpreter/encode.go

```diff
@@ -855,12 +855,13 @@ func (e *Encoder) encodeArray(
 const (
 	encodedDictionaryValueKeysFieldKey    uint64 = 0
 	encodedDictionaryValueEntriesFieldKey uint64 = 1
+	encodedDictionaryValueTypeFieldKey    uint64 = 2
```

Comment:
It's a bigger breaking change, but given that we have to perform a storage migration anyways, maybe we want to place the type info at the start, just to be consistent with arrays?

If that's too much work, then let's just leave it as-is

Reply:
I also like that. Shouldn't be a big change :+1: